### ISSUE TITLE
[Bug]: Voiceover is not announcing DetailsList group header name in Safari browser on Macbook

### ISSUE THREAD

**Issue by arpitgupta19** [microsoft/fluentui#26887]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.2
    CPU: (10) x64 Apple M1 Pro
    Memory: 16.28 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Browsers:
    Chrome: 109.0.5414.119
    Safari: 16.3

Works fine on chrome and edge
Doesn't work on safari.
```

### Are you reporting Accessibility issue?

Yes

### Reproduction

https://codepen.io/arpitgupta19/pen/XWPbeeq

## Actual Behavior
Voiceover in Safari just speaks '$MenuItemName, menuitem, group'

## Expected Behavior
Voiceover in Safari should speak '$MenuItemName, menuitem, group, **$GroupName**'

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
